Parsedown.php

```diff
@@ -553,15 +553,19 @@ protected function blockHeader($Line)
         }
 
         $text = trim($text, ' ');
+        $link = strtolower(str_replace(' ','-',$text));
```

Comment:
I think a more elaborate "sluggify" mechanism would be better here. For example, `strtolower(preg_replace('/[^A-z-_]+'/, '-', $text))` to replace all non A-z/-/_ characters with a dash.

Reply:
Yeah, I'm not sure what GitHub's method of replacement is but think we should probably be using:

```php
strtolower(preg_replace('/[^A-Za-z0-9\-_]+'/, '-', $text))
```

Would you agree with that one?

Reply:
https://regex101.com/r/jWgxEc/1 - This is an example but I also had to include \v to prevent new lines from being picked up in the example.

Reply:
You are right, this approach is a better start. GitHuv flavor looks quite complicated and forgiving, so I put my regex hat. 

Test gist: https://gist.github.com/Ayesh/250786888e7f4f146117aa96afcd0071
Suggested regex: [`[^\p{L}\p{N}\p{M}-]+`](https://regex101.com/r/hDI9DM/1)
Implementation: https://3v4l.org/melBh

The third heading in the Gist contains letters from my Sinhalese language. It has letters in both `\p{L}` (letters) and `\p{M}` (marks). These two, combined with `\p{N}` (numbers) and the dash gives us the negating regex that strips out everything else. 

This library already requires mbstring extension, so I don't see a problem with using `mb_strtolower`.

Reply:
OK, seems better 👍 I love it when a plan comes together.

Reply:
I have just pushed a change, which takes parts of yours but changes str_replace as well since we are talking potential multi-byte characters.  Seems to work with the gist, let me know what you think.